server/repository/migrations/postgres/2022-01-27T08-00_create_changelog_table/up.sql

```diff
@@ -18,47 +18,52 @@ CREATE TYPE row_action_type AS ENUM (
 );
 
 CREATE TABLE changelog (
-    id BIGSERIAL,
+    cursor BIGSERIAL,
```

Comment:
I'm not 100% on this, but I think you might want to add a `PRIMARY KEY` or `UNIQUE` keyword on this?

`Thus, we have created an integer column and arranged for its default values to be assigned from a sequence generator. A NOT NULL constraint is applied to ensure that a null value cannot be inserted. (In most cases you would also want to attach a UNIQUE or PRIMARY KEY constraint to prevent duplicate values from being inserted by accident, but this is not automatic.)`
- https://www.postgresql.org/docs/14/datatype-numeric.html

I'm not sure if you already considered this and didn't want the overhead of the unique checking, and duplication is ok(ish)?

Adding unique or primary_key will also create an index on that column, which should improve performance on the 
`changelog_deduped` view (especially when it get large)

Reply:
9a6ab624d0cd636fd58b98f4b6d1e020e1d25798

I'll double check in the `head` pr that it passes tests before merging (to make sure syntax is correct).

Thanks !

> I'm not sure if you already considered this and didn't want the overhead of the unique checking, and duplication is ok(ish)?

Oversight, I just renamed.

> Adding unique or primary_key will also create an index on that column, which should improve performance on the
changelog_deduped view (especially when it get large)

I agree, thanks for reminder, made an issue for [the rest of the schema](https://github.com/openmsupply/open-msupply/issues/603)